enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,642 @@
+will:
+1. Iterate through all involved components and configure them with non-conflicting configuration profiles (`config` verb).
+2. Await a high-level cluster steady state (cluster operators available, machines config up-to-date) if any configuration profile requires it.
+3. Run tests within the bucket with non-conflicting, test-level parallelization.
+
+#### Test Result Aggregation
+
+`openshift-tests` will store test results in the CI artifacts repository in a JSONL file.
+The output will be similar to that of the `run-test` output, but each test result 
+will be expanded to include:
+- Component & subcomponent
+- Environment
+- Testing Context
+
+The goal here is to allow a single file to contain comprehensive information necessary for
+a system like Component Readiness to analyze it without, for example, needing to draw information
+from prowjob names.
+
+```python
+{
+    "name": "test name",
+    
+    "result": "pass",
+
+    # ...other test result information...
+    
+    # Information necessary to fully qualify the 
+    # test.
+    "component": {
+        "product": "openshift",
+        "type": "payload",
+        "component": "hyperkube",
+        "subcomponent": "logs",
+    },
+    
+    "environment": {
+        "platform": "aws",
+        "architecture": "amd64",
+        # ...others...
+        
+        # Configuration information is also included. 
+        "configuration": {
+            # The configuration id applied to the component
+            # before the test was run.
+            "component": "default",
+        }
+    },
+    
+    # Information about how the test was run
+    "context": {
+        # A small number of seeds can be used to randomize test
+        # parallelization and bucketing. Analyzing seeds against
+        # one another should highlight parallelization issues.
+        # Using the same seed with the same tests should result
+        # in identical planning and parallelization. 
+        "seed": 15,
+        
+        # A sha256 of the tests IDs which were run. This, plus
+        # seed should ensure identical execution pattern between
+        # two runs.
+        "testHash": "deadbeef",
+        
+        # In the future, we may run tests several times against
+        # the same cluster in order to aggregate
+        # test results more quickly. We may want to 
+        # analyze results including this dimension if, for example,
+        # we want to check whether the first run is as consistent
+        # as the following N.
+        "run": 0
+    }
+}
+```
+
+### Risks and Mitigations
+
+#### Binary Incompatibility
+Copying a binary from a different image into a running pod exposes us to binary incompatibility. 
+
+##### CPU Architecture 
+An arm64 binary will not run in an amd64 based pod. `openshift-tests` will need to run on 
+a build farm node with a CPU architecture consistent with architecture of the cluster 
+payload under test.

```

Comment:
Or, use the multiarch payload

Reply:
This is harder to accomplish when dealing with CI payloads and pre-submit testing.